mkosi/__init__.py

```diff
@@ -2098,29 +2155,39 @@ def install_uki(context: Context, kver: str, kimg: Path, token: str, partitions:
         else:
             boot_binary = context.root / f"boot/EFI/Linux/{token}-{kver}{boot_count}.efi"
 
-    microcode = build_microcode_initrd(context)
-
-    initrds = [microcode] if microcode else []
-    initrds += context.config.initrds or [build_default_initrd(context)]
-
-    if context.config.kernel_modules_initrd:
-        initrds += [build_kernel_modules_initrd(context, kver)]
-
     # Make sure the parent directory where we'll be writing the UKI exists.
     with umask(~0o700):
         boot_binary.parent.mkdir(parents=True, exist_ok=True)
 
-    build_uki(
-        context,
-        systemd_stub_binary(context),
-        kver,
-        context.root / kimg,
-        initrds,
-        finalize_cmdline(context, roothash),
-        boot_binary,
-    )
+    if context.config.shim_bootloader == ShimBootloader.signed:
+        for p in (context.root / "usr/lib/modules" / kver).glob("*.efi"):
+            shutil.copy2(p, boot_binary)
+            break
```

Comment:
What's the reason for only picking the first one we find and not all of them? In the future there might be multiple different distro-provided UKIs that people could install. A debug log line about which UKI is picked here might be helpful.

Reply:
We'd have to figure out naming first, let's do that in a separate PR